presto-spi/src/main/java/io/prestosql/spi/type/AbstractIntType.java

```diff
@@ -64,6 +67,13 @@ public final Slice getSlice(Block block, int position)
     @Override
     public void writeLong(BlockBuilder blockBuilder, long value)
     {
+        if (value > Integer.MAX_VALUE) {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, format("Value %d exceeds MAX_INT", value));
+        }
+        if (value < Integer.MIN_VALUE) {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, format("Value %d is less than MIN_INT", value));
+        }
+
         blockBuilder.writeInt((int) value).closeEntry();
```

Comment:
toIntExact(value) ?

Reply:
The value is already verified explicitly above, so not needed. Also, intellij recognizes this and doesn't emit a warning here.

Reply:
👍

Reply:
I've seen that. Wasn't that on purpose (I guess not)?

Reply:
discussed offline